driver/sdk.md

```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
+
+
+### Configuring the build Dockerfile
+
+The building of the driver is done in a Docker container so you must adapt the
+Docker template `Dockerfile.build.tpl` that the `bblfsh` command created for you
+to install all the dependencies needed to build the `native` command detailed in
+the previous step.
+
+The templated dockerfile looks like this (without comments):
+
+```Dockerfile
+FROM alpine:3.5
+RUN mkdir -p /opt/driver/src && \
+    adduser ${BUILD_USER} -u ${BUILD_UID} -D -h /opt/driver/src
+RUN apk add --no-cache make git curl ca-certificates
+WORKDIR /opt/driver/src
+```
+
+As the comments in the file say, Alpine is the preferred distribution for 
+space reasons but you can use Debian slim or other slim versions if for any 
+reason Alpine don't work for your driver.
 
-The AST to UAST normalization is written in Go. The main file to be edited is
-`driver/normalizer/normalizer.go`.
+You should add any packages you need to the `apk add` line. Avoid adding more
+lines with more `RUN apk add` commands instead because that would create more
+intermediate layers (taking more space).
 
-TODO
+You can also install dependencies using the language package manager instead
+(pip, cargo, dub, npm, etc. if you prefer, after installing it with apk. 
+Another option is downloading and compiling the dependencies. If you go that
+way you'll probably need the `build-base` or `alpine-sdk` packages.
 
-## Building
+You can see a list and search for Alpine packages
+[here](https://pkgs.alpinelinux.org/packages).
+
+You don't need to run any command to build the `native` file; the SDK will 
+execute `make build-native-internal` target once the Docker has been created.
+
+To trigger the building and test your Dockerfile and building process you can
+issue the commands:
 
 ```bash
 $ bblfsh-sdk prepare-build
 $ make build
 ```
 
-TODO
+You'll see a sequence of `docker build` and `docker run` commands execute and any
+building error will show on the standard output.
+
+### Configuring the run Dockerfile
+
+The second dockerfile template that you have to edit is the one that the Babelfish
+server will use to run new instances of drivers. Like the build one it will also 
+be generated by the `bblfsh-sdk` command but with the filename `Dockerfile.tpl`
+and this initial content without comments:
+
+```Dockerfile
+FROM alpine:3.5
+CMD /opt/driver/bin/driver
+```
+
+As with the build container, the first step is to add any packages needed to
+**run** the `native` binary produced by the build process, if any is needed like
+dynamically loading libraries for compiled languages (not included in Alpine base)
+or runtimes for interpreted or compiled-to-bytecode languages.
+
+Then you will need to copy the `build` directory into `/opt/driver/bin`, so that
+both the (automatically build by the SDK) `driver` and `native` binaries will 
+rest in that directory.
 
-## Testing
+The final `CMD` command should remain at the end of the file and you should not
+change it.
```

Comment:
It'd be nice what the purpose of this `CMD` is, so that people understand why it should remain unchanged.